middleware/authentication.js

```diff
@@ -0,0 +1,23 @@
+const User = require("../models/User");
+const jwt = require("jsonwebtoken");
+const { UnauthenticatedError } = require("../errors");
+
+const authenticate = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (!authHeader || !authHeader.startsWith("Bearer")) {
+    throw new UnauthenticatedError("Authentication invalid");
+  }
+
+  const token = authHeader.split(" ")[1];
+
+  try {
+    const payload = jwt.verify(token, process.env.JWT_SECRET);
+    req.body = { ...req.body, userId: payload.userId, name: payload.userName, email: payload.userEmail };

```

Comment:
Since those values are more about the user identity and not the requests themselves, I suggest you add them to headers rather than the request body. That way you don't need to design your endpoints thinking about any conflicts between data.

Reply:
You could also follow the example from [CTD's repo](https://github.com/Code-the-Dream-School/node-express-course/blob/main/06-jobs-api/final/middleware/authentication.js) and set `req.user`.